src/plugins/data_source_management/public/components/data_source_table/data_source_table.tsx

```diff
@@ -250,10 +250,13 @@ export const DataSourceTable = ({ history }: RouteComponentProps) => {
 
   const setDefaultDataSource = async () => {
     try {
-      for (const dataSource of selectedDataSources) {
-        if (uiSettings.get('defaultDataSource') === dataSource.id) {
-          await setFirstDataSourceAsDefault(savedObjects.client, uiSettings, true);
-        }
+      const restDataSources = dataSources.filter(
+        (dataSource) => !selectedDataSources.includes(dataSource)
+      );
+      const restDataSourceIds = restDataSources.map((item) => item.id);
```

Comment:
nit: can we chain `filter()` and `map` commands, since I don't see `restDataSources ` being used elsewhere.

Reply:
Updated